test/helpers/conditions.go

```diff
@@ -0,0 +1,26 @@
+package helpers
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/samber/lo"
+
+	"github.com/kong/gateway-operator/pkg/consts"
+	k8sutils "github.com/kong/gateway-operator/pkg/utils/kubernetes"
+)
+
+// CheckAllConditionsTrue returns true if all the conditions with given type in `conditionTypes` are set to `True` in the given resource.
+// If it returns `false`, the second return value contains a message to tell what conditions are not `True`.
+func CheckAllConditionsTrue(resource k8sutils.ConditionsAware, conditionTypes []consts.ConditionType) (bool, string) {
+	failedConditions := make([]string, 0, len(conditionTypes))
+	lo.ForEach(conditionTypes, func(conditionType consts.ConditionType, _ int) {
+		if !k8sutils.HasConditionTrue(conditionType, resource) {
+			failedConditions = append(failedConditions, string(conditionType))
+		}
+	})
```

Comment:
Nit: I believe that `ForEach` here doesn't provide us any benefits (shorter syntax etc) so I'd be inclined to suggest to use a plain `for` loop. WDYT?

Reply:
Ping